### ISSUE TITLE
Tombstone support

### ISSUE THREAD

**Issue by promontis** [ch-robinson/dotnet-avro#32]
Hi @dstelljes! 

Maybe you could help us out... we are trying to produce a tombstone message to Kafka. From what I understand, it is not yet supported by onfluent-kafka-dotnet: see https://github.com/confluentinc/confluent-kafka-dotnet/issues/905.

We tried sending a null value to an avro topic that has a union of null and a record (the POCO), but the subscriber (a JDBC sink) doesn't seem to see this as a tombstone message, and crashes into a null-reference exception trying to deserialize the avro message. Therefore, I think that we shouldn't be sending the tombstone message as an avro serialized null value, but shortcircuit the serialization and sent a plain (non avro serialized) null, like the java implementation does: 

https://github.com/confluentinc/schema-registry/blob/master/avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java#L56

Is this currently possible with this library?

**Comment by dstelljes** [comment]
Yeah, that’s correct—with the Schema Registry serializers, you don’t get true null values. Your null/record union would be serialized as six bytes: the magic byte (position 0), the schema ID (1-4), and the union index (5).

You have a couple options right now:

1. Create two producer instances, one for regular messages and one for tombstones. You can use use `DependentProducerBuilder` to share a handle.

2. Create your own `ISerializer<TValue>` that returns a true null value when `null` is serialized, per https://github.com/confluentinc/confluent-kafka-dotnet/issues/905#issuecomment-486172105).

I’m uneasy about using the Java serializer’s short-circuit strategy by default, but adding a flag on the serializer to enable it might be something we can do.
